src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -0,0 +1,610 @@
+import com.avispl.symphony.dal.avdevices.encoderdecoder.haivision.x4decoder.dto.streamstats.StreamData;
+import com.avispl.symphony.dal.avdevices.encoderdecoder.haivision.x4decoder.dto.streamstats.StreamInfo;
+import com.avispl.symphony.dal.avdevices.encoderdecoder.haivision.x4decoder.dto.streamstats.StreamStats;
+import com.avispl.symphony.dal.communicator.RestCommunicator;
+import com.avispl.symphony.dal.util.StringUtils;
+
+/**
+ * An implementation of RestCommunicator to provide communication and interaction with Haivision X4 Decoders
+ * Supported features are:
+ * <p>
+ * Monitoring:
+ * <li>DeviceInfo</li>
+ * <li>DecoderStats</li>
+ * <li>StreamStats</li>
+ * <p>
+ * Controlling:
+ * <li>Start/Stop /Edit Decoder</li>
+ * <li>Create/ Edit/ Delete Stream</li>
+ *
+ * @author Harry
+ * @since 1.0.0
+ */
+public class HaivisionX4DecoderCommunicator extends RestCommunicator implements Monitorable, Controller {
+
+	private AuthenticationCookie authenticationCookie;
+	private AuthenticationInfo authenticationInfo;
+	private HashMap<String, String> failedMonitor;
+	private ObjectMapper objectMapper;
+	private Set<String> streamNameSet;
+	private Set<String> streamStatusSet;
+	private Set<String> portNumberSet;
+
+	//Adapter Properties
+	private String streamName;
+	private String portNumber;
+	private String streamStatus;
+
+	/**
+	 * Retrieves {@code {@link #streamName }}
+	 *
+	 * @return value of {@link #streamName}
+	 */
+	public String getStreamName() {
+		return streamName;
+	}
+
+	/**
+	 * Sets {@code streamsName}
+	 *
+	 * @param streamsName the {@code java.lang.String} field
+	 */
+	public void setStreamName(String streamsName) {
+		this.streamName = streamsName;
+	}
+
+	/**
+	 * Retrieves {@code {@link #portNumber}}
+	 *
+	 * @return value of {@link #portNumber}
+	 */
+	public String getPortNumber() {
+		return portNumber;
+	}
+
+	/**
+	 * Sets {@code portNumber}
+	 *
+	 * @param portNumber the {@code java.lang.String} field
+	 */
+	public void setPortNumber(String portNumber) {
+		this.portNumber = portNumber;
+	}
+
+	/**
+	 * Retrieves {@code {@link #streamStatus}}
+	 *
+	 * @return value of {@link #streamStatus}
+	 */
+	public String getStreamStatus() {
+		return streamStatus;
+	}
+
+	/**
+	 * Sets {@code streamStatus}
+	 *
+	 * @param streamStatus the {@code java.lang.String} field
+	 */
+	public void setStreamStatus(String streamStatus) {
+		this.streamStatus = streamStatus;
+	}
+
+	/**
+	 * This method is called by Symphony to control the properties in the device
+	 *
+	 * @param controllableProperty ControllableProperty instance
+	 */
+	@Override
+	public void controlProperty(ControllableProperty controllableProperty) {
+		//TODO
+	}
+
+	/**
+	 * This method is called by Symphony to control the properties in the device
+	 *
+	 * @param list the list ControllableProperty instance
+	 */
+	@Override
+	public void controlProperties(List<ControllableProperty> list) {
+		// TODO
```

Comment:
is it gonna be this version or the next one?

Reply:
It is gonna be this version, let me describe more info about it.